### PR TITLE
[KEYCLOAK-11415] Switch the 'GroupMapperConfig.PRESERVE_GROUP_INHERITANCE' setting reliably

### PR DESCRIPTION
**Pull request by iankko**
Use own, separate context when trying to switch 'GroupMapperConfig.PRESERVE_GROUP_INHERITANCE'
    group mapper config setting to 'false' (or back), across the various tests from LDAPGroupMapperSyncTest
    suite. This makes the test results deterministic again (prevents 'test02_syncWithGroupInheritance()'
    and 'test03_syncWithDropNonExistingGroups()' tests randomly to fail depending if attempt
    to reset the 'GroupMapperConfig.PRESERVE_GROUP_INHERITANCE' back to 'true' in previous
    'test01_syncNoPreserveGroupInheritance()' test succeeded, or not)
    
    Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

### CODE REVIEW

**Comment by iankko** [comment]
@mposolda PTAL
@pskopek JFYI -^

**Comment by iankko** [comment]
An example of a failure, this should fix is [here](https://travis-ci.org/keycloak/keycloak/jobs/584540661).

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/federation/ldap/LDAPGroupMapperSyncTest.java [hunk 1]
```diff
@@ -239,6 +266,11 @@ public void test03_syncWithDropNonExistingGroups() throws Exception {
             ComponentModel mapperModel = LDAPTestUtils.getSubcomponentByName(realm, ctx.getLdapModel(), "groupsMapper");
             LDAPStorageProvider ldapProvider = LDAPTestUtils.getLdapProvider(session, ctx.getLdapModel());
 
+            // KEYCLOAK-11415 - This test requires the group mapper to be configured with preserve group inheritance
+            // set to 'true' (the default setting). If preservation of group inheritance isn't configured, some of
+            // the previous test(s) failed to cleanup properly. Check the requirement as part of running the test
+            Assert.assertEquals(mapperModel.getConfig().getFirst("preserve.group.inheritance"), "true");
```
Comment (hmlnarik) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/federation/ldap/LDAPGroupMapperSyncTest.java]:
```suggestion
            Assert.assertEquals(mapperModel.getConfig().getFirst(GroupMapperConfig.PRESERVE_GROUP_INHERITANCE
), "true");
```

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/federation/ldap/LDAPGroupMapperSyncTest.java [hunk 2]
```diff
@@ -192,6 +213,11 @@ public void test02_syncWithGroupInheritance() throws Exception {
             LDAPStorageProvider ldapProvider = LDAPTestUtils.getLdapProvider(session, ctx.getLdapModel());
             GroupLDAPStorageMapper groupMapper = LDAPTestUtils.getGroupMapper(mapperModel, ldapProvider, realm);
 
+            // KEYCLOAK-11415 - This test requires the group mapper to be configured with preserve group inheritance
+            // set to 'true' (the default setting). If preservation of group inheritance isn't configured, some of
+            // the previous test(s) failed to cleanup properly. Check the requirement as part of running the test
+            Assert.assertEquals(mapperModel.getConfig().getFirst("preserve.group.inheritance"), "true");
```
Comment (hmlnarik) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/federation/ldap/LDAPGroupMapperSyncTest.java]:
```suggestion
            Assert.assertEquals(mapperModel.getConfig().getFirst(GroupMapperConfig.PRESERVE_GROUP_INHERITANCE), "true");
```

**Comment by hmlnarik** [comment]
To switch the `GroupMapperConfig.PRESERVE_GROUP_INHERITANCE` setting reliably, you should better use `try { .. } finally { ..}` instead

**Review by mposolda — APPROVED** [review]
I agree with the suggestions from @hmlnarik . However considering that we're in a hurry with this, suggestions are minor and travis successfully passed, I am approving the PR.

**Pull request merged**